src/Bridge/Symfony/Bundle/EventListener/EventDispatcher.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Symfony\Bundle\EventListener;
+
+use Symfony\Component\EventDispatcher\Event;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+
+/**
+ * @author Alan Poulain <contact@alanpoulain.eu>
+ */
+final class EventDispatcher
+{
+    private $eventName;
+    private $eventClass;
+    private $dispatcher;
+
+    public function __construct(string $eventName, string $eventClass, EventDispatcherInterface $dispatcher)
+    {
+        $this->eventName = $eventName;
+        $this->eventClass = $eventClass;
+        $this->dispatcher = $dispatcher;
+    }
+
+    public function dispatch(Event $event): void
+    {
+        $internalEvent = null;
+
+        switch ($event) {
+            case $event instanceof GetResponseEvent:
+                $internalEvent = new $this->eventClass(null, ['request' => $event->getRequest()]);
+        }
+
+        $this->dispatcher->dispatch($this->eventName, $internalEvent);
```

Comment:
if the internal event is null, how about dispatching the given event instead ?

Reply:
Why would it be null? I'm thinking of throwing an exception instead (it would mean the service is not configured correctly).

Reply:
As this service is kinda decorating the event dispatcher, if you want to dispatch other events, you will dispatch a `null` event instead. I think this can be a huge source of bugs or miscomprehension.

Unless this service is really for internal...

Reply:
Yes I think it's really internal. If an user want to do something like this, it can do it itself. Maybe adding a `@internal` for the class would be a good idea.